File: backend/src/main/java/com/inveny/backend/entity/Stock.java
package com.inveny.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Generated;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Stock {
    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    private Long id;
    @Column(
            nullable = false
    )
    private String name;
    private Integer quantity;
    private String unit;
    private String category;
    private String memo;
    private String location;

}

File: frontend/lib/models/stock.dart
class Stock {
  final int? id;
  final String name;
  final int quantity;
  final String unit;
  final String? category;
  final String location;
  final String? memo;

  Stock({
    this.id, 
    required this.name, 
    required this.quantity, 
    required this.unit,
    this.category,
    required this.location,
    this.memo
    });

  factory Stock.fromJson(Map<String, dynamic> json) {
    return Stock(
      id: json['id'],
      name: json['name'],
      quantity: json['quantity'],
      unit: json['unit'],
      category: json['category'],
      location: json['location'] ?? '주방',
      memo: json['memo'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'quantity': quantity,
      'unit': unit,
      'category': category,
      'location': location,
      'memo': memo,
    };
  }
}